Sources/BraintreeLocalPayment/BTLocalPaymentRequest.swift

```diff
@@ -11,55 +11,73 @@ import BraintreeDataCollector
-    
-    /// Optional: Payer email of the customer.
-    public var email: String?
-    
-    /// Optional: Given (first) name of the customer.
-    public var givenName: String?
-    
-    /// Optional: Surname (last name) of the customer.
-    public var surname: String?
-    
-    /// Optional: Phone number of the customer.
-    public var phone: String?
-    
-    ///  Indicates whether or not the payment needs to be shipped. For digital goods, this should be false. Defaults to false.
-    public var isShippingAddressRequired: Bool = false
+    // MARK: - Internal Properties
     
-    /// Optional: Bank Identification Code of the customer (specific to iDEAL transactions).
-    public var bic: String?
+    let paymentType: String?
+    let paymentTypeCountryCode: String?
+    let amount: String?
+    let merchantAccountID: String?
+    let address: BTPostalAddress?
+    let currencyCode: String?
+    let displayName: String?
+    let email: String?
+    let givenName: String?
+    let surname: String?
+    let phone: String?
+    var isShippingAddressRequired: Bool = false
+    let bic: String?
     
     public weak var localPaymentFlowDelegate: BTLocalPaymentRequestDelegate?
-    
-    // MARK: - Internal Properties
 
     var paymentID: String?
     var correlationID: String?
+    
+    // MARK: - Initializer
+
+    /// Creates a LocalPaymentRequest
+    /// - Parameters:
+    ///   - paymentType: The type of payment.
+    ///   - paymentTypeCountryCode: The country code of the local payment. This value must be one of the supported country codes for a given local payment type listed at the link below.
+    ///   For local payments supported in multiple countries, this value may determine which banks are presented to the customer.
+    ///     - SeeAlso: https://developer.paypal.com/braintree/docs/guides/local-payment-methods/client-side-custom/ios/v5#invoke-payment-flow
+    ///   - amount: The amount for the transaction.
+    ///   - merchantAccountID: Optional: A non-default merchant account to use for tokenization.
+    ///   - address: Optional: The address of the customer. An error will occur if this address is not valid.
+    ///   - currencyCode: Optional: A valid ISO currency code to use for the transaction. Defaults to merchant currency code if not set.
+    ///   - displayName: Optional: The merchant name displayed inside of the local payment flow.
+    ///   - email: Optional: Payer email of the customer.
+    ///   - givenName: Optional: Given (first) name of the customer.
+    ///   - surname: Optional: Surname (last name) of the customer.
+    ///   - phone: Optional: Phone number of the customer.
+    ///   - isShippingAddressRequired: Indicates whether or not the payment needs to be shipped. For digital goods, this should be `false`. Defaults to `false`.
+    ///   - bic: Optional: Bank Identification Code of the customer (specific to iDEAL transactions).
+    public init(
+        paymentType: String? = nil,
```

Comment:
is everything _really_ optional?

Reply:
Thanks for asking, i've updated it. `paymentType`, `amount` and `currencyCode` are required for all LPMs, others depend on the type.
As docs says: `paymentTypeCountryCode` is required for payment methods with multiple potential country codes, otherwise it will use the `countryCode` ([reference](https://developer.paypal.com/braintree/docs/guides/local-payment-methods/client-side-custom/javascript/v3/#render-local-payment-method-buttons))